packages/builder/src/build.js

```diff
@@ -140,11 +140,6 @@ async function copyManifest(manifestJSON) {
       }
     }
 
-    copy.commands = manifestJSON.commands.map(command => {
-      const basename = path.basename(command.script)
```

Comment:
Ah now I remember why I was only taking the basename: if you have a command script that is for example `../../something.js`, it won't be in the resulting plugin.

I guess the best solution would be to have a better transformation of the path: maybe `replace(/[./]/g, '_')`?

Reply:
Ah, I see! Doing a straight-up replacement won’t cover the case where there’s both `foo/bar` and `foo_bar` though. We could use a “special” character (or characters) that’s less common than `_`, which would probably cover 99% of cases?

Another solution is to make the paths all be relative to the closest common parent folder? Example:

```
└── code
    ├── foo
    │   └── bar.js
    └── my-plugin
        └── baz.js
```

`my-plugin` is our plugin folder, and `code` is the most common parent folder:
- `../foo/bar` command maps to `foo/bar`
- `./baz` command maps to `my-plugin/baz`

What do you think?

(Should absolute paths be supported as well?)

Reply:
It's true that `foo/bar` and `foo_bar` will create a conflict but I'd say it'd be super rare? In that case we can show a warning as well? But yes we can choose a less common character.

The algo for your second solution becomes a lot more complicated I think. I'd rather go with a dumb solution for now.

As for absolute path, I'd say it's a really bad practice because it will break for everybody else

Reply:
Pushed some changes:
1. Use a simple replace to flatten the path. The regex is `/\.(?!js)|\//g` so as to not mangle the file extension.
2. Added some logic to detect conflicts and throw an error if it happens – do agree that it’s quite unlikely, as you mentioned.